src/Codeception/Module/MongoDb.php

```diff
@@ -114,6 +106,70 @@ public function _initialize()
         }
     }
 
+    private function validateDump()
+    {
+        if ($this->config['dump'] && ($this->config['cleanup'] or ($this->config['populate']))) {
+            if (!file_exists(Configuration::projectDir() . $this->config['dump'])) {
+                throw new ModuleConfigException(
+                    __CLASS__,
+                    "File with dump doesn't exist.\n
+                    Please, check path for dump file: " . $this->config['dump']
+                );
+            }
+            $this->dumpFile = Configuration::projectDir() . $this->config['dump'];
+            $this->isDumpFileEmpty = false;
+
+            switch ($this->config['dump_type']) {
```

Comment:
this switch-case seems to be too complicated. I'd suggest to transform it to using single `if` (try to avoid `else`) and moving complex logic to other methods, like `validateJSDump`, `validateMongoDump`, etc.